packages/block-library/src/navigation-link/edit.js

```diff
@@ -78,6 +80,36 @@ function NavigationLinkEdit( {
 		}
 	}, [ isSelected ] );
 
+	// If the LinkControl popover is open and the URL has changed, close the LinkControl and focus the label text.
+	useEffect( () => {
+		if ( isLinkOpen && url ) {
+			// Close the link.
+			setIsLinkOpen( false );
+
+			// Does this look like a URL and have something TLD-ish?
+			if ( isURL( prependHTTP( label ) ) && /^.+\.[a-z]+/.test( label ) ) {
+				// Focus and select the label text.
+				selectLabelText();
+			} else {
+				// Focus it (but do not select).
+				placeCaretAtHorizontalEdge( ref.current, true );
+			}
+		}
+	}, [ url ] );
+
+	/**
+	 * Focus the navigation link label text and select it.
+	 */
+	function selectLabelText( ) {
```

Comment:
I feel we already have a way to do this using selectors/actions without messing with refs, selection... right @ellatrix ?

Reply:
Yes, if you have access to the value (if HTML you need to convert to rich text value), the block client ID and attribute key.

Reply:
Why does it need to be selected?

Reply:
It's not about getting the value, but to actually select/highlight the text so that someone can start typing delete the label value. This was discussed here: https://github.com/WordPress/gutenberg/pull/19686#issuecomment-575280635

Can I focus the input on the DOM and highlight/select it without using `ref`s? I'm very new to React and Gutenberg, so any direction is helpful and appreciated :)